### PR TITLE
Add `DeleteClusterMaintenanceConfig` for terraform

### LINKED ISSUE #26185: Can't delete cluster_maintenance_config after creating
Expected behavior:

A `cluster_maintenance_config` is removable by `tctl rm cluster_maintenance_config/cluster_maintenance_config`.

Current behavior:

After creating `cluster_maintenance_config` you cannot remove it.  A cluster administrator may need to remove automatic upgrades or was created in error.

```bash
$ tctl rm cluster_maintenance_config/cluster_maintenance_config
ERROR: deleting resources of type "cluster_maintenance_config" is not supported
```

Bug details:
- Teleport version: 13.0.0
- Recreation steps:

1. Create a `cluster_maintenance_config` configuration
2. Attempt to remove with `tctl rm cluster_maintenance_config/cluster_maintenance_config`

### PR DESCRIPTION
**Pull request by lxea**
The terraform code gen in teleport-plugins expects a Delete method for resources

Part of #26066
Fixes #26185

### CODE REVIEW

File: tool/tctl/common/resource_command.go [hunk 1]
```diff
@@ -1042,6 +1043,11 @@ func (rc *ResourceCommand) Delete(ctx context.Context, client auth.ClientI) (err
 			return trace.Wrap(err)
 		}
 		fmt.Printf("cluster auth preference has been reset to defaults\n")
+	case types.KindClusterMaintenanceConfig:
+		if err := client.DeleteClusterMaintenanceConfig(ctx); err != nil {
```
Comment (greedy52) [inline tool/tctl/common/resource_command.go:1047]: any consequences for deleting it? I see we are doing a reset for other configs instead.

Comment (lxea) [reply tool/tctl/common/resource_command.go:1047]: The config isnt created by default so resetting it to an empty version doesnt seem right, but I'm not sure what the correct behavior should be here

Comment (greedy52) [reply tool/tctl/common/resource_command.go:1047]: As long as the caller is ok with not found. Seems fine:
https://github.com/gravitational/teleport/blob/64d3e8c505c11159ab795fc33d4138b6632fc5db/lib/auth/auth.go#L825-L833

**Review by greedy52 — APPROVED** [review]

**Review by r0mant — COMMENTED** [review]
@fspmarshall Can you review this please? Is it ok to let self-hosted users delete cloud maintenance config? This is needed to let terraform manage the resource.

**Review by fspmarshall — APPROVED** [review]

**Comment by fspmarshall** [comment]
Deleting cmc is totally fine, just needs to be disabled for Cloud, which this PR does.

**Pull request merged**